### PR TITLE
feat: check user permissions and consider feature flag on home page

### PR DESCRIPTION
**Pull request by ruzniaievdm**
[2U-155](https://youtrack.raccoongang.com/issue/2U-155)

### CODE REVIEW

File: src/studio-home/card-item/index.jsx [hunk 1]
```diff
@@ -1,16 +1,26 @@
...
 }) => {
+  const {
+    allowCourseReruns,
+    courseCreatorStatus,
+    rerunCreatorStatus,
+  } = useSelector(getStudioHomeData);
   const courseUrl = new URL(url, getConfig().STUDIO_BASE_URL);
   const subtitle = isLibraries ? `${org} / ${number}` : `${org} / ${number} / ${run}`;
+  // eslint-disable-next-line max-len
+  const isShowRerunLink = allowCourseReruns && rerunCreatorStatus && courseCreatorStatus === COURSE_CREATOR_STATES.granted;
```
Comment (PKulkoRaccoonGang) [inline src/studio-home/card-item/index.jsx:23]: Maybe 😄 
```suggestion
  const isShowRerunLink = allowCourseReruns 
  && rerunCreatorStatus 
  && courseCreatorStatus === COURSE_CREATOR_STATES.granted;
```

File: src/studio-home/collapsible-state-with-action/index.jsx [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    },
+    disabledStates: [STATEFUL_BUTTON_STATES.pending, STATEFUL_BUTTON_STATES.error],
+  };
+
+  // eslint-disable-next-line no-nested-ternary
+  const requestButtonCurrentState = courseCreatorSavingStatus === RequestStatus.PENDING
+    ? STATEFUL_BUTTON_STATES.pending
+    : courseCreatorSavingStatus === RequestStatus.FAILED
+      ? STATEFUL_BUTTON_STATES.error
+      : STATEFUL_BUTTON_STATES.default;
```
Comment (PKulkoRaccoonGang) [inline src/studio-home/collapsible-state-with-action/index.jsx:47]: Nested ternary expressions are hard to read. I propose to create constants and simplify this code.

File: src/studio-home/home-sidebar/index.jsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    platformName,
+    studioShortName,
+    studioRequestEmail,
+    techSupportEmail,
+    courseCreatorStatus,
+  } = useSelector(getStudioHomeData);
+  const { home: aboutHomeLink } = useHelpUrls(['home']);
+
+  // eslint-disable-next-line max-len
+  const isShowMailToGetInstruction = courseCreatorStatus === COURSE_CREATOR_STATES.disallowedForThisSite && !!studioRequestEmail;
```
Comment (PKulkoRaccoonGang) [inline src/studio-home/home-sidebar/index.jsx:25]: Maybe 😄 
```suggestion
  const isShowMailToGetInstruction = courseCreatorStatus === COURSE_CREATOR_STATES.disallowedForThisSite 
  && !!studioRequestEmail;
```

File: src/studio-home/home-sidebar/index.jsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    <HelpSidebar>
+      <h4 className="help-sidebar-about-title">
+        {intl.formatMessage(messages.aboutTitle, { studioName })}
+      </h4>
+      <p className="help-sidebar-about-descriptions">
+        {intl.formatMessage(messages.aboutDescription, { studioShortName })}
+      </p>
+      <HelpSidebarLink
+        as="span"
+        pathToPage={aboutHomeLink || ''}
```
Comment (PKulkoRaccoonGang) [inline src/studio-home/home-sidebar/index.jsx:39]: Also usage keys for `BLANK_STRING`

File: src/studio-home/processing-courses/index.jsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+import { getStudioHomeData } from '../data/selectors';
+import CourseItem from './course-item';
+import messages from './messages';
+
+const ProcessingCourses = () => {
+  const intl = useIntl();
+  const { inProcessCourseActions } = useSelector(getStudioHomeData);
+
+  return (
+    <div>
```
Comment (PKulkoRaccoonGang) [inline src/studio-home/processing-courses/index.jsx:15]: Maybe with fragment will be better
```suggestion
    <></>
```

File: src/studio-home/scss/StudioHome.scss [hunk 6]
```diff
@@ -85,3 +85,10 @@
     margin: 0;
   }
 }
+
+.spinner-icon {
+  animation-name: rotate;
+  animation-duration: 2000ms;
+  animation-iteration-count: infinite;
+  animation-timing-function: linear;
```
Comment (PKulkoRaccoonGang) [inline src/studio-home/scss/StudioHome.scss:93]:
```suggestion
animation: rotate 2s infinite linear;
```

File: src/course-rerun/course-rerun-form/index.jsx [hunk 7]
```diff
@@ -5,7 +5,7 @@ import { useIntl } from '@edx/frontend-platform/i18n';
 import { CreateOrRerunCourseForm } from '../../generic/create-or-rerun-course';
 import messages from './messages';
 
-const CourseRerunForm = ({ initialFormValues, handleOnClickCancel, handleOnClickCreate }) => {
```
Comment (vladislavkeblysh) [inline src/course-rerun/course-rerun-form/index.jsx:8]: optional: onClickCancel, onClickCreate

File: src/generic/create-or-rerun-course/CreateOrRerunCourseForm.jsx [hunk 8]
```diff
@@ -192,7 +227,7 @@ const CreateOrRerunCourseForm = ({
             key={field.label}
           >
             <Form.Label>{field.label}</Form.Label>
-            {!field.isDropdown ? (
+            {field.name !== 'org' ? (
```
Comment (vladislavkeblysh) [inline src/generic/create-or-rerun-course/CreateOrRerunCourseForm.jsx:230]: maybe create const with this string?

File: src/studio-home/StudioHome.jsx [hunk 9]
```diff
@@ -1,53 +1,74 @@
 import React from 'react';
 import {
-  Button, Container, Layout,
+  Button, Container, Layout, MailtoLink,
```
Comment (vladislavkeblysh) [inline src/studio-home/StudioHome.jsx:3]: optional: 

Button, 
Container, 
Layout, 
MailtoLink,

File: src/studio-home/card-item/index.jsx (hunk 1 above)
Comment (ruzniaievdm) [reply src/studio-home/card-item/index.jsx:23]: ok, fine)

File: src/studio-home/home-sidebar/index.jsx (hunk 3 above)
Comment (ruzniaievdm) [reply src/studio-home/home-sidebar/index.jsx:25]: ok, fine)

File: src/studio-home/processing-courses/index.jsx (hunk 5 above)
Comment (ruzniaievdm) [reply src/studio-home/processing-courses/index.jsx:15]: yeah, sure!

File: src/course-rerun/course-rerun-form/index.jsx (hunk 7 above)
Comment (ruzniaievdm) [reply src/course-rerun/course-rerun-form/index.jsx:8]: changed

File: src/generic/create-or-rerun-course/CreateOrRerunCourseForm.jsx (hunk 8 above)
Comment (ruzniaievdm) [reply src/generic/create-or-rerun-course/CreateOrRerunCourseForm.jsx:230]: I think it's gonna be a little redundant.

File: src/studio-home/collapsible-state-with-action/index.jsx (hunk 2 above)
Comment (ruzniaievdm) [reply src/studio-home/collapsible-state-with-action/index.jsx:47]: changed

**Review by vladislavkeblysh — APPROVED** [review]

**Review by wowkalucky — APPROVED** [review]
OK

**Pull request merged**